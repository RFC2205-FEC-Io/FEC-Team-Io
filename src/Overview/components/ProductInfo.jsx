import React from 'react';
import Stars from './Stars.jsx';
import SocialMedia from './SocialMedia.jsx';
class ProductInfo extends React.Component {
  constructor (props) {
    super (props);
    this.state = {
      products: {},
      price: 0,
      salePrice: 0,
      show: false,
      reviews: []
    }
    this.updatePrice = this.updatePrice.bind(this);
  }

  componentDidMount () {
    this.setState({
      products: this.props.products,
      price: this.props.defaultPrice,
      salePrice: this.props.salePrice,
      reviews: this.props.reviews
    })
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevState.products !== this.props.products) {
      this.setState({products: this.props.products});
    }
    if (prevState.price !== this.props.defaultPrice) {
      this.setState({price: this.props.defaultPrice});
    }
    if (prevState.salePrice!== this.props.salePrice) {
      this.setState({salePrice: this.props.salePrice});
    }
    if (prevState.products !== this.props.products) {
      this.setState({reviews: this.props.reviews});
    }
  }

  updatePrice () {
    if (this.state.salePrice === null || this.state.salePrice === 0) {
      return (
        <div>
          <p id='category'>{this.state.products.category}</p>
          <h2 id='name'>{this.state.products.name}</h2>
          <p id='default-price'>{this.state.price}</p>
          <p id='slogan'>{this.state.products.slogan}</p>
        </div>
      )
    } else {
      return (
        <div>
          <p id='category'>{this.state.products.category}</p>
          <h2 id='name'>{this.state.products.name}</h2>
          <p id='price'>{this.state.price}</p>
          <p id='salePrice'>{this.state.salePrice}</p>
        </div>
      )
    }
  }

  render (props) {
    return (
      <div id='product-info'>
        {this.updatePrice()}
        <Stars
          reviews={this.state.reviews}
        />
        <SocialMedia/>
      </div>
    );
  }
}

export default ProductInfo;


