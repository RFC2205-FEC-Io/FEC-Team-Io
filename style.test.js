// import 'react-native';
import React from 'react';
import {shallow} from 'enzyme';
import ProductInfo from '/Users/lawrence/FEC-Team-Io/src/Overview/components/ProductInfo.jsx';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

configure({ adapter: new Adapter() });
describe('first snapshot test', () => {
  it ('renders the AddToCart component correctly', () => {
    const wrapper = shallow(<ProductInfo/>);
    expect(wrapper).toMatchSnapshot();
  });
});

