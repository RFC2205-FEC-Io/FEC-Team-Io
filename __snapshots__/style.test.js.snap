// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first snapshot test renders the AddToCart component correctly 1`] = `ShallowWrapper {}`;
